samples/webassembly/Triangle/Triangle.cpp

```diff
@@ -0,0 +1,254 @@
+  renderPipelineState_Triangle_ = device_->createRenderPipeline(desc, nullptr);
+  IGL_ASSERT(renderPipelineState_Triangle_);
+}
+
+static std::shared_ptr<ITexture> getNativeDrawable() {
+  Result ret;
+  std::shared_ptr<ITexture> drawable;
+  const auto& platformDevice = device_->getPlatformDevice<opengl::webgl::PlatformDevice>();
+  IGL_ASSERT(platformDevice != nullptr);
+  drawable = platformDevice->createTextureFromNativeDrawable(&ret);
+
+  IGL_ASSERT_MSG(ret.isOk(), ret.message.c_str());
+  IGL_ASSERT(drawable != nullptr);
+  return drawable;
+}
+
+static void createFramebuffer(const std::shared_ptr<ITexture>& nativeDrawable) {
+  FramebufferDesc framebufferDesc;
+  framebufferDesc.colorAttachments[0].texture = nativeDrawable;
+
+  const TextureDesc desc = TextureDesc::new2D(
+      nativeDrawable->getFormat(),
+      nativeDrawable->getDimensions().width,
+      nativeDrawable->getDimensions().height,
+      TextureDesc::TextureUsageBits::Attachment | TextureDesc::TextureUsageBits::Sampled,
+      IGL_FORMAT("{}C{}", framebufferDesc.debugName.c_str(), 0).c_str());
+
+  framebufferDesc.colorAttachments[1].texture = device_->createTexture(desc, nullptr);
+  framebuffer_ = device_->createFramebuffer(framebufferDesc, nullptr);
+  IGL_ASSERT(framebuffer_);
+}
+
+static void render(const std::shared_ptr<ITexture>& nativeDrawable) {
+  const auto size = framebuffer_->getColorAttachment(0)->getSize();
+  if (size.width != width_ || size.height != height_) {
+    createFramebuffer(nativeDrawable);
+  } else {
+    framebuffer_->updateDrawable(nativeDrawable);
+  }
+
+  // Command buffers (1-N per thread): create, submit and forget
+  CommandBufferDesc cbDesc;
+  std::shared_ptr<ICommandBuffer> buffer = commandQueue_->createCommandBuffer(cbDesc, nullptr);
+
+  const igl::Viewport viewport = {0.0f, 0.0f, (float)width_, (float)height_, 0.0f, +1.0f};
+  const igl::ScissorRect scissor = {0, 0, (uint32_t)width_, (uint32_t)height_};
+
+  // This will clear the framebuffer
+  auto commands = buffer->createRenderCommandEncoder(renderPass_, framebuffer_);
+
+  commands->bindRenderPipelineState(renderPipelineState_Triangle_);
+  commands->bindViewport(viewport);
+  commands->bindScissorRect(scissor);
+  commands->pushDebugGroupLabel("Render Triangle", igl::Color(1, 0, 0));
+  commands->draw(PrimitiveType::Triangle, 0, 3);
+  commands->popDebugGroupLabel();
+  commands->endEncoding();
+
+  buffer->present(nativeDrawable);
+
+  commandQueue_->submit(*buffer);
+}
+
+void emscriptenMainLoopCallback() {
+  render(getNativeDrawable());
+  glfwPollEvents();
+}
+
+int main(int argc, char* argv[]) {
+  renderPass_.colorAttachments.resize(1);
+  initWindow(&window_);
+  initIGL();
+
+  createFramebuffer(getNativeDrawable());
+  createRenderPipeline();
+
+  // Main loop
+  emscripten_set_main_loop(&emscriptenMainLoopCallback, 0, 1);
+
+  renderPipelineState_Triangle_ = nullptr;
+  framebuffer_ = nullptr;
+  device_.reset(nullptr);
```

Comment:
Missing reset for `commandQueue_` before the device.

Reply:
Also, please use consistent style.
Either all should be ` = nullptr;` or all should be `reset(...)`